File: movemork/Features/Auth/AuthProofSummaryCard.swift
//
//  AuthProofSummaryCard.swift
//  movemork
//
//  Prominent proof receipt — the thing the user is saving.
//

import SwiftUI

struct AuthProofSummaryCard: View {
    let mode: AuthContainerView.Mode

    var body: some View {
        MMProofReceiptCard(
            statusLabel: statusLabel,
            title: "Kitchen proof",
            metaLine: metaLine,
            thumbnailImageName: "welcome-kitchen-main"
        )
        .id("auth-summary-\(mode == .signIn ? "in" : "up")")
    }

    private var statusLabel: String {
        mode == .signIn ? "Sign in to save proof" : "Create account to save proof"
    }

    private var metaLine: String {
        mode == .signIn ? "Your move-in photos stay in your vault." : "Start with one room — add more anytime."
    }
}

File: movemark-android/app/src/main/java/com/surensureshkumar/movemark/features/auth/AuthProofSummaryCard.kt
package com.surensureshkumar.movemark.features.auth

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.surensureshkumar.movemark.R
import com.surensureshkumar.movemark.core.design.components.MMProofArtifactCard
import com.surensureshkumar.movemark.core.design.components.MMProofArtifactModel
import com.surensureshkumar.movemark.core.design.components.MMProofStatusTone

@Composable
fun AuthProofSummaryCard(
    mode: AuthMode,
    modifier: Modifier = Modifier,
) {
    val statusLabel = if (mode == AuthMode.SignIn) {
        "Sign in to save proof"
    } else {
        "Create account to save proof"
    }
    val metaLine = if (mode == AuthMode.SignIn) {
        "Your move-in photos stay in your vault."
    } else {
        "Start with one room — add more anytime."
    }

    MMProofArtifactCard(
        model = MMProofArtifactModel(
            phaseEyebrow = "Room Proof",
            phaseLabel = "Move-in",
            roomName = "Kitchen proof",
            photoCount = 2,
            issueCount = 1,
            verifiedLabel = metaLine,
            savedLabel = statusLabel,
            statusLabel = statusLabel,
            statusTone = MMProofStatusTone.Success,
        ),
        drawableResId = R.drawable.welcome_kitchen_main,
        photoHeight = 120.dp,
        cornerRadius = 18.dp,
        tagsVisible = false,
        modifier = modifier,
    )
}
